kodein-type/src/jvmTest/kotlin/org/kodein/type/JavaType.kt

```diff
@@ -32,4 +33,18 @@ class JavaType {
         }
     }
 
+    @Test fun test03_genericArrayType() {
+        // Test generic<>()
+        assertTrue(generic<Array<List<String>>>() is JVMGenericArrayTypeToken)
+        // Test raw Type
+        assertEquals(List::class.java, (generic<Array<List<String>>>().getRaw() as JVMClassTypeToken).jvmType)
+        assertEquals(List::class.java, (generic<Array<List<String>>>().jvmType as ParameterizedType).rawType)
+        // Test generic parameters (when Array<ParameterizedType>)
+        assertEquals(String::class.java, generic<Array<List<String>>>().getGenericParameters()[0].jvmType)
+        assertEquals(String::class.java, ((generic<Array<List<String>>>().jvmType as ParameterizedType).actualTypeArguments[0] as WildcardType).upperBounds[0])
+        // Test typeToken of GenericArrayType -> Java primitive array
+        assertEquals(Byte::class.java, typeToken(GenericArrayType { java.lang.Byte.TYPE }).jvmType)
+        assertEquals(Int::class.java, typeToken(GenericArrayType { Integer.TYPE }).jvmType)
+    }
```

Comment:
There needs to be more tests.
How do you represent a generic array type using erased ?

We need a way to express the type `Array<List<String>>` using erased. Maybe create an `erasedArray` function?

There need to be tests that validates that:

```
assertEquals(generic<Array<List<String>>>, erasedArray(erasedComp(List::class, String::class)))
```

By the way, there is absolutely no reason those tests are JVM only.